projects/angular-grid-layout/src/lib/utils/grid.utils.ts

```diff
@@ -15,9 +15,7 @@ export function ktdTrackById(index: number, item: {id: string}) {
  * @param cols, number of columns of the grid.
  */
 export function ktdGridCompact(layout: KtdGridLayout, compactType: KtdGridCompactType, cols: number): KtdGridLayout {
-    return compact(layout, compactType, cols)
-        // Prune react-grid-layout compact extra properties.
-        .map(item => ({id: item.id, x: item.x, y: item.y, w: item.w, h: item.h}));
+    return compact(layout, compactType, cols);
```

Comment:
`compact` function is entirely taken from 'react-grid-layout' and it comes with extra properties that we don 't want for now, like:

```
    moved?: boolean;
    static?: boolean;
    isDraggable?: boolean | null | undefined;
    isResizable?: boolean | null | undefined;
```

 So I would suggest following the previous approach adding the min and max properties. You can make a function and use it on all places that is needed.